extension/src/codeLensContainer.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { ConfigurationChangeEvent, Disposable, languages, workspace, WorkspaceConfiguration } from 'vscode';
+import { testCodeLensProvider } from './codeLensProvider';
+
+const JAVA_TEST_SETTINGS_CONFIGURATION: string = 'java.test.settings';
+const ENABLE_CODE_LENS_VARIABLE: string = 'enableTestCodeLens';
+const LANGUAGE: string = 'java';
+
+class TestCodeLensContainer implements Disposable {
+  private lensProvider: Disposable | undefined;
+  private configurationEvent: Disposable;
+
+  constructor() {

```

Comment:
Though register the listeners in the constructor can make it work here. I still prefer to using a method called `initialize()` and call it explicitly instead of put listener registration in the constructor.